src/transformers/models/gpt_neo/modeling_gpt_neo.py

```diff
@@ -1121,7 +1121,7 @@ def forward(
                     f"unexpected if using padding tokens in conjunction with `inputs_embeds.`"
                 )
 
-        pooled_logits = logits[range(batch_size), sequence_lengths]
+        pooled_logits = logits[torch.arange(batch_size), sequence_lengths]
```

Comment:
I think you can use `torch.take_along_dim(logits, sequence_lengths, dim=1)`. You might need to match shape of logits and sequence_lengths. 

=> It will remove the need to gather from the shape object.

Reply:
I applied the following changes:

```python
          if isinstance(sequence_lengths, torch.Tensor):
              pooled_logits = torch.take_along_dim(
                  logits, indices=sequence_lengths.unsqueeze(1).unsqueeze(1), dim=1
              ).squeeze()
          else:
              pooled_logits = logits[torch.arange(batch_size), sequence_lengths]

```
The forward pass works and give the same output as original version, but ONNX conversion fails with:
`RuntimeError: Exporting the operator take_along_dim to ONNX opset version 11 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.`.